api/src/main/java/com/datastrato/gravitino/exceptions/IllegalPropertyException.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datastrato.gravitino.exceptions;
+
+import com.google.errorprone.annotations.FormatMethod;
+import com.google.errorprone.annotations.FormatString;
+
+/** An exception thrown when a property is invalid. */
+public class IllegalPropertyException extends GravitinoRuntimeException {
```

Comment:
@jerryshao Since I added the `IllegalPropertyException` and it's not a subclass of `IllegalArgumentException`, I'm not sure if I should maintain the compatibility of this interface or modify the interface signature:
https://github.com/apache/gravitino/blob/178eb37f8b5013abdd2464dda764ddd5b0787f38/api/src/main/java/com/datastrato/gravitino/SupportsCatalogs.java#L107-L108

Reply:
I removed this, plz help to review when you have time, thanks!